linter/ginkgo_linter.go

```diff
@@ -1651,6 +1793,31 @@ func isExprErrFunc(pass *analysis.Pass, expr ast.Expr) bool {
 	return false
 }
 
+// Returns whether the expression is a function that takes in a single gomega.Gomega argument and
+// returns nothing, e.g. `func(g gomega.Gomega) { g.Expect(myStringMaker()).ToNot(BeEmpty()) }`
+func isExprGomegaFunc(pass *analysis.Pass, expr ast.Expr) bool {
+	actualArgType := pass.TypesInfo.TypeOf(expr)
+	t, isFunc := actualArgType.(*gotypes.Signature)
+	if !isFunc {
+		return false
+	}
+
+	if t.Results().Len() != 0 || t.Params().Len() != 1 {
+		return false
+	}
```

Comment:
Continue: this condition will give the wrong result for valid usages like:

```go
Eventually(func(g Gomega, ctx context.Context){...})...
```
Or
```go
Eventually(func(g Gomega, x, y int){...}).WithArguments(1,2)...
```